CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.6)
+
+if (WIN32 AND DEFINED ENV{VCPKG_ROOT})
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
+endif ()
+
+set(n clinfo)
+project(${n})
+
+find_package(OpenCL REQUIRED)
+
+add_executable(${n} src/${n}.c)
+target_include_directories(${n} PUBLIC src)
+target_include_directories(${n} PUBLIC ${OpenCL_INCLUDE_DIRS})
+target_link_libraries(${n} ${OpenCL_LIBRARIES})
+
+if (WIN32)
```

Comment:
The property is not Windows specific but generator specific. If you want to guard it, correct check is
```cmake
if (CMAKE_GENERATOR MATCHES "Visual Studio")
```
However, I would omit the check, as this directory property has no effect on generators other than VS. Perhaps a comment might suffice. If you really would like to cater to IDE users, I'd also add source_group properties for Filters in the project.

Reply:
I've omitted the check. I'll leave you to suggest a source_group properties addition. I think they look ugly on small projects, and we don't need complexity at this stage in negotiations with the repo owner.